### PR TITLE
add functionality for dual farkas on variables bound

### PR DESCRIPTION
**Pull request by joaquimg**
cc @odow

### CODE REVIEW

**Comment by joaquimg** [comment]
can we merge this and merge test right after?
so that @IssamT can work on MOI 0.6

File: src/solve.jl [hunk 1]
```diff
@@ -164,15 +165,20 @@ end
...
+        set = MOI.get(model, MOI.ConstraintSet(), index)
+        primal_value = model.variable_primal_solution[column]
+        if is_binding(set, primal_value)
+            return model.variable_dual_solution[column]
+        else
+            return 0.0
+        end
     end
-end
+ end
```
Comment (odow) [inline src/solve.jl]: Nit: space

File: src/solver_interface.jl [hunk 2]
```diff
@@ -521,6 +521,18 @@ each linear constraint.
...
 
+"""
+    get_farkas_dual_bounds!(m, x::Vector{Float64})
+
+Get the farkas dual (certificate of primal infeasiblility) for the variable
+bounds in the model `m`, and store in `x`. `x`must have one element for
+each variable.
+Since most solvers do not have this feature, this funciton has a default fallback that does nothing.
+"""
+function get_farkas_dual_bounds!(m::LinQuadOptimizer, x)
```
Comment (odow) [inline src/solver_interface.jl]: Style: single letter variable names

File: src/solver_interface.jl [hunk 3]
```diff
@@ -521,6 +521,18 @@ each linear constraint.
 """
 function get_farkas_dual! end
 
+"""
+    get_farkas_dual_bounds!(m, x::Vector{Float64})
+
+Get the farkas dual (certificate of primal infeasiblility) for the variable
+bounds in the model `m`, and store in `x`. `x`must have one element for
+each variable.
+Since most solvers do not have this feature, this funciton has a default fallback that does nothing.
```
Comment (odow) [inline src/solver_interface.jl]: s/funciton/function. Also long line length.

**Comment by joaquimg** [comment]
Perfect!
Thanks

**Pull request merged**